dvc/tree/s3.py

```diff
@@ -199,12 +199,13 @@ def isfile(self, path_info):
         return path_info.path in self._list_paths(path_info)
 
     def _list_paths(self, path_info, max_items=None):
-        kwargs = {"Prefix": path_info.path}
-        if max_items is not None:
-            kwargs["MaxKeys"] = max_items
-
         with self._get_bucket(path_info.bucket) as bucket:
-            for obj_summary in bucket.objects.filter(**kwargs):
+            obj_summaries = bucket.objects.filter(Prefix=path_info.path)
+            if max_items is not None:
+                obj_summaries = obj_summaries.page_size(max_items).limit(
```

Comment:
Do we even need to alter page size anymore?

Reply:
If we don't add it, it will try to fetch a list 1000 summaries and return the first one AFAIK.

Reply:
You can also try it with the code that I gave in the description that is also much faster with `page_size()`. Runs with `page_size()`;
```
 $ python t.py
Took:  2.2795308610002394
(.venv38) (Python 3.8.5+) [ 11:53ÖÖ ]  [ isidentical@desktop:~/temp_dvc(master✗) ]
 $ python t.py
Took:  2.2522677979995933
(.venv38) (Python 3.8.5+) [ 11:53ÖÖ ]  [ isidentical@desktop:~/temp_dvc(master✗) ]
 $ python t.py
Took:  2.1077582990001247
```
Runs without `page_size()`;
```
 $ python t.py
Took:  3.382246265999129
(.venv38) (Python 3.8.5+) [ 11:53ÖÖ ]  [ isidentical@desktop:~/temp_dvc(master✗) ]
 $ python t.py
Took:  3.211584769998808
(.venv38) (Python 3.8.5+) [ 11:53ÖÖ ]  [ isidentical@desktop:~/temp_dvc(master✗) ]
 $ python t.py
Took:  5.401903420999588
```

Reply:
Thanks for clarifying! :pray: